pkg/apis/jenkins.io/v1/types_extensions.go

```diff
@@ -438,6 +438,40 @@ type AppList struct {
 	Items []App `json:"items"`
 }
 
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// Application is the metadata for an Application
+type Application struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ApplicationSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ApplicationList is a structure used by k8s to store lists of apps
+type ApplicationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []Application `json:"items"`
+}
+
+// ApplicationSpec provides details of the metadata for an App
+type ApplicationSpec struct {
+	Name        string
```

Comment:
what is name here?  (or how would it be distinct from the actual resource name)? (or maybe I misunderstand this)

Reply:
Yeah, not entirely sure what I was thinking. I can't see how it would be different from `Repo` in my mind.